src/string/immutable.zig

```diff
@@ -1119,6 +1125,16 @@ pub fn index(self: string, str: string) i32 {
     }
 }
 
+/// Returns a substring starting at `start` up to the end of the string.
+/// If `start` is greater than the string's length, returns an empty string.
+pub fn drop(self: string, start: usize) string {
```

Comment:
Can we name this something like `trimStart`, `trimPrefix`, `stripStart`, or `stripPrefix`? `drop` could lead one to believe it frees memory.

Reply:
This has been a large point of contention in the internal discord and we ended up on no agreement.

It seems to be a split

<img width="406" height="302" alt="image" src="https://github.com/user-attachments/assets/a5aa8469-f9c0-4881-8888-b64976983014" />

I think the old way is the worst of all but @Jarred-Sumner and @nektro disagree. I don't want to be gerrymandering the voting system but I am open to @pfgithub's suggestion if it means we move away from the legacy repetitive noisy way.

Likely for now I'll just revert the existing implementation to the legacy way in order to get this PR merged.

Reply:
If we've settled on `drop`, would `dropStart` or `dropLeft` be acceptable? If I see `s.drop(3)` my first assumption is that it removes characters from the *end* of the string, like `pop`.

Reply:
For context: http://www.zvon.org/other/haskell/Outputprelude/drop_f.html

Reply:
I think I've made my case by authority here as much as I am willing to. It's not that big of a deal and I really care about this PR meeting 1.3 so I will do as per @Jarred-Sumner's suggestion -- `foo[@min(foo.len, 3)..]` just to ensure this doesn't get blocked for a pretty minor debate.

Reply:
I want to do this as a future improvement, if that's OK, since time is running out

Reply:
I stand with @pfgithub's proposal of using `.substring` with the JS semantics. Although I dislike it personally, it's miles ahead of using `[@min(...)]` inline.